frontends/PyCDE/src/CMakeLists.txt

```diff
@@ -6,42 +6,79 @@
 #
 # ===-----------------------------------------------------------------------===//
 
+include(AddMLIRPython)
+
+add_compile_definitions("MLIR_PYTHON_PACKAGE_PREFIX=pycde.circt.")
+
 declare_mlir_python_sources(PyCDESources
   ROOT_DIR "${CMAKE_CURRENT_SOURCE_DIR}"
   SOURCES
-  pycde/__init__.py
-  pycde/pycde_types.py
-  pycde/dialects/__init__.py
-  pycde/dialects/msft.py
-  pycde/dialects/seq.py
-  pycde/dialects/hw.py
-  pycde/dialects/comb.py
-  pycde/dialects/sv.py
-  pycde/dialects/fsm.py
-  pycde/dialects/hwarith.py
-  pycde/support.py
-  pycde/module.py
-  pycde/behavioral.py
-  pycde/constructs.py
-  pycde/common.py
-  pycde/system.py
-  pycde/devicedb.py
-  pycde/instance.py
-  pycde/value.py
-  pycde/ndarray.py
-  pycde/esi.py
-  pycde/esi_api.py
-  pycde/esi_runtime_common.py
-  pycde/fsm.py
-  pycde/testing.py
-
-  pycde/esi_api.py.j2
-  pycde/Makefile.cosim
+  __init__.py
+  pycde_types.py
+  dialects/__init__.py
+  dialects/msft.py
+  dialects/seq.py
+  dialects/hw.py
+  dialects/comb.py
+  dialects/sv.py
+  dialects/fsm.py
+  dialects/hwarith.py
+  support.py
+  module.py
+  behavioral.py
+  constructs.py
+  common.py
+  system.py
+  devicedb.py
+  instance.py
+  value.py
+  ndarray.py
+  esi.py
+  esi_api.py
+  esi_runtime_common.py
+  fsm.py
+  testing.py
+
+  esi_api.py.j2
+  Makefile.cosim
+)
+
+################################################################################
+# Build composite binaries
+################################################################################
+set(PYCDE_PYTHON_PACKAGE_DIR "${CIRCT_PYTHON_PACKAGES_DIR}/pycde/")
+
+# Bundle our own, self-contained CAPI library with all of our deps.
+add_mlir_python_common_capi_library(PyCDE_CIRCTPythonCAPI
+  INSTALL_COMPONENT CIRCTBindingsPythonModules
+  INSTALL_DESTINATION python_packages/pycde/pycde/circt/_mlir_libs
+  OUTPUT_DIRECTORY "${PYCDE_PYTHON_PACKAGE_DIR}/pycde/circt/_mlir_libs"
+  RELATIVE_INSTALL_ROOT "../../.."
+  DECLARED_SOURCES
+    MLIRPythonSources.Core
+    CIRCTBindingsPythonExtension
+)
+
+add_mlir_python_modules(PyCDE_CIRCTPythonModules
```

Comment:
Does this need to be duplicated in PyCDE's CMakeLists? I would have been nice to reuse the original CIRCTPythonModules, maybe with some tweaks to the flags passed to add_mlir_python_modules. But I assume doing it this way was required.

Reply:
The current `CIRCTPythonModules` puts everything in `circt_core/circt`, not `pycde/pycde/circt`.